### DISCUSSION TITLE
v7.4.1 Release of PowerShell

### DISCUSSION THREAD

**Discussion by SeeminglyScience** [PowerShell/PowerShell#21050] [General]
### General Cmdlet Updates and Fixes

- Fix `Group-Object` output using interpolated strings (#20745) (Thanks @mawosoft!)
- Fix `Start-Process -PassThru` to make sure the `ExitCode` property is accessible for the returned `Process` object (#20749) (#20866) (Thanks @CodeCyclone!)
- Fix rendering of `DisplayRoot` for network PSDrive (#20793) (#20863)

### Engine Updates and Fixes

- Ensure filename is not null when logging WDAC ETW events (#20910) (Thanks @jborean93!)
- Fix four regressions introduced by WDAC audit logging feature (#20913)

### Build and Packaging Improvements

<details>

<summary>

Bump .NET 8 to version 8.0.101

</summary>

<ul>
<li>Update .NET SDK and dependencies for v7.4.1 (Internal 29142)</li>
<li>Update cgmanifest for v7.4.1 (#20874)</li>
<li>Update package dependencies for v7.4.1 (#20871)</li>
<li>Set the <code>rollForwardOnNoCandidateFx</code> in <code>runtimeconfig.json</code> to roll forward only on minor and patch versions (#20689) (#20865)</li>
<li>Remove RHEL7 publishing to packages.microsoft.com as it's no longer supported (#20849) (#20864)</li>
<li>Fix the tab completion tests (#20867)</li>
</ul>

</details>

[7.4.1]: https://github.com/PowerShell/PowerShell/compare/v7.4.0...v7.4.1

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - C593CDCF4585547DC3F1F5B9373F1998ADA71360B42CC03E1C890F663507830C
- powershell_7.4.1-1.deb_amd64.deb
  - 625B7EE0B71147421723CB6022A41B5D8FC0D6E19DF25B1240008EE491BF6997
- powershell-7.4.1-1.cm.aarch64.rpm
  - 40D75F7B00CA57E978E76A10E09FFA906DA8996667CB28572331EA9A4B6FCE84
- powershell-7.4.1-1.cm.x86_64.rpm
  - D59DF093FCCC6571E1D30D2963FFD73EF89DE290EA11A3EBB9722E870B7E7905
- powershell-7.4.1-1.rh.x86_64.rpm
  - 1D34921F600A03D92E30FD1582A6B25C34E7F6D1324CB50289F6522EE78EC015
- powershell-7.4.1-linux-arm32.tar.gz
  - BA66844392026779145C10A72D9B9096FE9F71F196AECFFCE45678C25D37C007
- powershell-7.4.1-linux-arm64.tar.gz
  - 719C11544A1F6322F2C58F5591FF7ABB7768AEF97EF0A382E3711F89FA2ADE22
- powershell-7.4.1-linux-musl-x64.tar.gz
  - 95B6287FE18F114A3833CEE917D89DC3B5D98C7405200AC13CFCCABDF4793522
- powershell-7.4.1-linux-x64-fxdependent.tar.gz
  - 2FFCD09C5CDD28FD2ED8574D138FFB3D522E6ADBBE30F1AAB236A92359C34D63
- powershell-7.4.1-linux-x64-musl-noopt-fxdependent.tar.gz
  - 0EC48F47741A7BCF6789CFD10DC7A55A4CE12843B26B8902E4DAFB7FF1BFC840
- powershell-7.4.1-linux-x64.tar.gz
  - 8BEABAC4431B75E9B67C6D9FAFD8B02CC019F21F230EA3D94AEC1348C50529D3
- powershell-7.4.1-osx-arm64.pkg
  - FA6C4499FB36A851F72F6136CE2D6804DC5C50E5BDEDBE0207B6FC60C68712F4
- powershell-7.4.1-osx-arm64.tar.gz
  - 435D135943F0F1E3A66FEEFA0251D73E9E8076F512DE9C469CDB205C5BE71967
- powershell-7.4.1-osx-x64.pkg
  - 6279B3E81406FD7B6124FF5AFAE211DE1E6CCE46568749874E4FF2BB64B78EDC
- powershell-7.4.1-osx-x64.tar.gz
  - 1FADEAE0C634D28F0E208493BFB5419D265DF057702C4BA80B212B644D97DC00
- PowerShell-7.4.1-win-arm64.zip
  - 086FFC7A2CEF14B3412F4240AFAED075F5BB6DAFE3090DA61C72A39E3171A47C
- PowerShell-7.4.1-win-fxdependent.zip
  - E26BED9E0791960E5D5BD6BECD714F318AA01A0B8F6B3BA26F9856AAB6773038
- PowerShell-7.4.1-win-fxdependentWinDesktop.zip
  - FD4CD7F19F5706ECC58C0E8FD9BE786D61C8D801A3CF9F8A559389D4331D5000
- PowerShell-7.4.1-win-x64.msi
  - 66C7C35ED9A46BD27E3D915DCF9A05E38B3F5EBB039883B92AA62FFEA20FB187
- PowerShell-7.4.1-win-x64.zip
  - 32DFA0D6B59538E28631549F04D3F74F3D3A24BC6EAF3AA7AB0B74F94ECC026E
- PowerShell-7.4.1-win-x86.msi
  - A8F034A8547415043C7C9905D64123F74F85F100B6AF391C4F33C3A45D113F90
- PowerShell-7.4.1-win-x86.zip
  - 55F6CA98C8BEDC1436E0B16A6A5522CAF68F2D7B27312B640DEE071E3596E3AF
- powershell-lts_7.4.1-1.deb_amd64.deb
  - 0916F1B7AD3F56E9DCADAE1A7B22B0FA28D4542D3AAAE50FE3979DE1C13C1FD2
- powershell-lts-7.4.1-1.cm.aarch64.rpm
  - 7EED2CB28DBD180CF2C34086E0CE73359FDFD47C4DA324DBA325A6C2429F0CE0
- powershell-lts-7.4.1-1.cm.x86_64.rpm
  - DECFC148C98008B00DAE10028EA9601F34841F796A6B4C2E004A33D86E33BB40
- powershell-lts-7.4.1-1.rh.x86_64.rpm
  - 3D5453043E6391F8AD57DD043B38BC46DFF7D1E665E38E98A5BC418C770B7DF3
- powershell-lts-7.4.1-osx-arm64.pkg
  - 2C0C240ECF69BE1FA38EF47A05BB31E45C0AE33B18B6AFAF8D90838EC1BDAAD3
- powershell-lts-7.4.1-osx-x64.pkg
  - F2E47A4D762A1522A7AD38E67D2353A59F0B29718B15179B4068D292D8535749

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.4.1'>v7.4.1 Release of PowerShell</a>.</em>

**Comment by MrJia1997** [comment]
No `.msixbundle` again ??

**Reply by ShreyasJejurkar** [reply]
cc @adityapatwardhan

**Comment by rhubarb-geek-nz** [comment]
It was three years ago that I started building ARM PowerShell packages for my Raspberry Pi. No ARM linux packages except for Mariner, osx ARM64 package still requires Rosetta and no Windows MSI for ARM64.

[So here they are](https://github.com/rhubarb-geek-nz/PowerShell/releases/tag/7.4.1).

**Comment by 237dmitry** [comment]
Please, add a man page in future releases. To avoid users having to write this themselves:

![ss](https://github.com/PowerShell/PowerShell/assets/78153320/45c86d4e-0e93-4c14-82d4-97a87bc655ef)

**Reply by rhubarb-geek-nz** [reply]
If you installed either the rpm or deb you would get the man page.

`/usr/local/share/man/man1/pwsh.1.gz`

```
$ man pwsh
PWSH(1)                                                                                                       PWSH(1)

NAME
       pwsh - PowerShell command-line shell and .NET REPL

SYNOPSIS
       pwsh  [-Login] [ [-File] filePath [args] ] [-Command { - | script-block [-args arg-array] | string [CommandPa‐
       rameters] } ] [-ConfigurationFile filePath] [-ConfigurationName string] [-CustomPipeName string]  [-EncodedAr‐
       guments  Base64EncodedArguments]  [-EncodedCommand  Base64EncodedCommand]  [-ExecutionPolicy  ExecutionPolicy]
       [-Help] [-InputFormat {Text | XML}] [-Interactive] [-MTA] [-NoExit] [-NoLogo]  [-NonInteractive]  [-NoProfile]
       [-NoProfileLoadTime]  [-OutputFormat {Text | XML}] [-SettingsFile filePath] [-SSHServerMode] [-STA] [-Version]
       [-WindowStyle

                                                     January 2024                                             PWSH(1)
```

Are you saying it is incomplete? If so I agree and simplest would be a "SEE ALSO" which gives a URL to the pages on the net.

**Reply by 237dmitry** [reply]
> Are you saying it is incomplete?

No, I am about absence. There is not man page in official tar.gz distribution.

```
$ (dir $PSHome/pwsh* -s).Name
pwsh
pwsh.deps.json
pwsh.dll
pwsh.runtimeconfig.json
pwsh.xml

```

**Reply by rhubarb-geek-nz** [reply]
The tars/zips are the contents of the binary runtime, say, /opt/microsoft/powershell/7 . The man pages don't belong in that directory, they live under /usr/local/share/man/man1 when installed by the rpm or deb.

**Reply by 237dmitry** [reply]
> they live under /usr/local/share/man/man1 when installed by the rpm or deb.

I don't use deb or rpm. And I never will. I am a home user, and after 20 years of using Linux as a desktop system, I have developed my own preferences in this regard.

**Reply by rhubarb-geek-nz** [reply]
To get the manual page download the debian package, use `ar` to extract `data.tar.gz` then use `tar` to extract `./usr/local/share/man/man1/pwsh.1.gz`

**Reply by 237dmitry** [reply]
> To get the manual page download the debian package

It's not even funny. Users stop whatever they are doing and look for man pages in the deb, rpm or wherever. Personally, I just formatted the output of `pwsh --help`. Not that I really needed it, but rather for order. Please don't argue out of nowhere, just agree that having `pwsh.1.gz` would be logical, regardless of how pwsh is distributed. And it would be better to have a more complete manual than `pwsh --help`.

**Reply by rhubarb-geek-nz** [reply]
I suggest you are the exception. If you install the deb or rpm package directly downloaded from here, or from microsoft.packages.com then you will get the man page installed in the standard place for man to find. You will also get the dependencies resolved and /etc/shells updated.

**Reply by 237dmitry** [reply]
> If you install the deb or rpm package

I don't use dpkg, apt, yum or dnf. It's in the AUR and I could have installed pwsh using paru, but I liked tar.gz as a more flexible way. It’s easy for me to add an entry to `/etc/shells` and create a symlink myself to pwsh.1.gz in `/usr/share/man/man1`

```powershell
sudo /usr/bin/echo '/usr/bin/pwsh' >> /etc/shells
sudo ln -s /opt/microsoft/powershell/7/pwsh.1.gz /usr/share/man/man1/pwsh.1.gz
```

**Reply by rhubarb-geek-nz** [reply]
It is a shame that the packaged `manpage` does not look anything like [pwsh.1.ronn](https://github.com/PowerShell/PowerShell/blob/master/assets/pwsh.1.ronn)

**Reply by rhubarb-geek-nz** [reply]
I would suggest that the manpage(s) goes in its(their) own separate tar file that is common to all architectures. The existing tar files contain the architecture specific binaries that live in /opt/microsoft/powershell/7 and the manpage tar contains what would go under /usr/local/share/man. Then you can choose where to put them.

**Reply by 237dmitry** [reply]
Yes, this would be the best.

**Comment by SupSkiFun** [comment]
Thank you for this release.  Any ETA when this will be available as a SNAP?  As of UTC Thursday, January 18, 2024 6:37:33 PM it does not exist at https://snapcraft.io/powershell.

**Reply by SupSkiFun** [reply]
Showing available as of UTC Friday, January 19, 2024 3:05:26 AM (likely sooner - just noticed it now).  Running well on Ubuntu 23.10 as a SNAP.  Thanks.

**Comment by Micael2030** [comment]
Hay que descargar un archivo solo; o todos los correspondientes para windows?

**Reply by rhubarb-geek-nz** [reply]
Sólo necesitas uno. Sugiero el archivo msi x64

**Comment by rhubarb-geek-nz** [comment]
Stats from my [ARM/ARM64 packages](https://github.com/rhubarb-geek-nz/PowerShell/releases/tag/7.4.1)

| Package | Downloads |
| ---------- | ----------- |
| powershell_7.4.1-1.deb_armhf.deb | 1 |
| powershell_7.4.1-1.deb_arm64.deb | 2 |
| powershell-7.4.1-1.rh.aarch64.rpm| 3 |
| powershell-7.4.1-osx-arm64.pkg | 2 |
| PowerShell-7.4.1-win-arm64.msi | 55 |
